fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1705,4 +1710,22 @@ private void officeSpecificLoanProductValidation(final Long productId, final Lon
         }
     }
 
+    private void validateTransactionProcessingStrategy(final LoanProduct loanProduct, final String loanRepaymentStrategy) {
+        // PROGRESSIVE: Repayment strategy MUST be only "advanced payment allocation"
+        if (loanProduct.getLoanProductRelatedDetail().getLoanScheduleType().equals(LoanScheduleType.PROGRESSIVE)) {
```

Comment:
Still no good :(

The loan can override the loan schedule type, so you must check that one instead of the loan product.

Reply:
The only check you need to do with product:
- if loan product repayment strategy is **not** advanced payment allocation, during the loan submit you **cannot** select advanced payment allocation either.